plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoHiveCatalog.java

```diff
@@ -319,6 +324,16 @@ public String defaultTableLocation(ConnectorSession session, SchemaTableName sch
         if (useUniqueTableLocation) {
             tableNameForLocation += "-" + randomUUID().toString().replace("-", "");
         }
+        if (database.getLocation().isEmpty() || database.getLocation().get().isEmpty()) {
+            if (warehouse == null) {
+                throw new TrinoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location cannot be determined, " +
+                        "please either set 'location' when creating the database, or set 'iceberg.catalog.warehouse' " +
+                        "to allow a default location at 'warehousePath/databaseName.db'", database.getDatabaseName()));
+            }
+            database = Database.builder(database)
```

Comment:
Let's not build a fake Database object. 
Instead, let's modify `getTableDefaultLocation` (which already has related logic).